src/components/generic/Button.tsx

```diff
@@ -0,0 +1,37 @@
+interface ButtonProps {
+    type: 'start' | 'pause' | 'resume' | 'reset' | 'fastforward';
+    onClick?: () => void;
+    disabled?: boolean;
+}
+
+const Button: React.FC<ButtonProps> = ({type, onClick, disabled = false }) => {
+    const buttonStyles = {
```

Comment:
I would probably merge these two objects so that everything about the `start` is in the same place. Something like:
```
start: {
  label: ...
  style: ...
}
```